From 60b8729b0a5670c1e484c4120eba653878d13186 Mon Sep 17 00:00:00 2001
From: Baptiste Robert <baptistex.robert@intel.com>
Date: Wed, 7 Aug 2013 11:32:45 +0200
Subject: Mixer: Check validity of input value

BZ: 129089
[CLEAN]

When the user wants to set a mixer with a value greater
than the maximum or smaller than the minimum, the mixer
is set at a ramdom value.

This patch set purpose is to return a error if the input
value is out of range.

Category: aosp improvement
Domain: AudioComms-Tinyalsa
Origin: internal
Upstream-Candidate: yes
Change-Id: Iee7fcfccfe78ce3196b77ed1938b0f22efdd6bc3
Orig-Change-Id: I3ec4d17a8f4b4baf88b256dfd61b34f23cba8b27
Signed-off-by: Baptiste Robert <baptistex.robert@intel.com>
---
 mixer.c |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/mixer.c b/mixer.c
index 5b9d3df..f074ed5 100644
--- a/mixer.c
+++ b/mixer.c
@@ -407,6 +407,11 @@ int mixer_ctl_set_value(struct mixer_ctl *ctl, unsigned int id, int value)
         break;
 
     case SNDRV_CTL_ELEM_TYPE_INTEGER:
+        if ((value < mixer_ctl_get_range_min(ctl)) ||
+            (value > mixer_ctl_get_range_max(ctl))) {
+            return -EINVAL;
+        }
+
         ev.value.integer.value[id] = value;
         break;
 
-- 
1.7.4.1

